solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -141,6 +134,16 @@ public int getId() {
     return id;
   }
 
+  /**
+   * @param id
+   *          Unique ID for this feature. Similar to feature name, except it can
+   *          be used to directly access the feature in the global list of
+   *          features.
+   */
+  public void setId(int id) {
```

Comment:
The name is an actual part of the feature (and its json) but the id is indirect i.e. calculated as the index of the feature in the store. So I kept id out of the constructor signature, at least for now, with a view towards seeing if later perhaps we can get rid of Feature.id somehow.